pandas/tests/indexes/multi/test_get_set.py

```diff
@@ -0,0 +1,649 @@
+# -*- coding: utf-8 -*-
+
+
+import numpy as np
+import pandas as pd
+import pandas.util.testing as tm
+import pytest
+from pandas import CategoricalIndex, Index, IntervalIndex, MultiIndex
+from pandas.compat import lrange, range
+from pandas.core.indexes.base import InvalidIndexError
+from pandas.util.testing import assert_almost_equal
+
+
+def test_get_loc(idx):
+    assert idx.get_loc(('foo', 'two')) == 1
+    assert idx.get_loc(('baz', 'two')) == 3
+    pytest.raises(KeyError, idx.get_loc, ('bar', 'two'))
+    pytest.raises(KeyError, idx.get_loc, 'quux')
+
+    pytest.raises(NotImplementedError, idx.get_loc, 'foo',
+                  method='nearest')
+
+    # 3 levels
+    index = MultiIndex(levels=[Index(lrange(4)), Index(lrange(4)), Index(
+        lrange(4))], labels=[np.array([0, 0, 1, 2, 2, 2, 3, 3]), np.array(
+            [0, 1, 0, 0, 0, 1, 0, 1]), np.array([1, 0, 1, 1, 0, 0, 1, 0])])
+    pytest.raises(KeyError, index.get_loc, (1, 1))
+    assert index.get_loc((2, 0)) == slice(3, 5)
+
+
+def test_get_loc_duplicates():
+    index = Index([2, 2, 2, 2])
+    result = index.get_loc(2)
+    expected = slice(0, 4)
+    assert result == expected
+    # pytest.raises(Exception, index.get_loc, 2)
+
+    index = Index(['c', 'a', 'a', 'b', 'b'])
+    rs = index.get_loc('c')
+    xp = 0
+    assert rs == xp
+
+
+def test_get_loc_level():
+    index = MultiIndex(levels=[Index(lrange(4)), Index(lrange(4)), Index(
+        lrange(4))], labels=[np.array([0, 0, 1, 2, 2, 2, 3, 3]), np.array(
+            [0, 1, 0, 0, 0, 1, 0, 1]), np.array([1, 0, 1, 1, 0, 0, 1, 0])])
+
+    loc, new_index = index.get_loc_level((0, 1))
+    expected = slice(1, 2)
+    exp_index = index[expected].droplevel(0).droplevel(0)
+    assert loc == expected
+    assert new_index.equals(exp_index)
+
+    loc, new_index = index.get_loc_level((0, 1, 0))
+    expected = 1
+    assert loc == expected
+    assert new_index is None
+
+    pytest.raises(KeyError, index.get_loc_level, (2, 2))
+
+    index = MultiIndex(levels=[[2000], lrange(4)], labels=[np.array(
+        [0, 0, 0, 0]), np.array([0, 1, 2, 3])])
+    result, new_index = index.get_loc_level((2000, slice(None, None)))
+    expected = slice(None, None)
+    assert result == expected
+    assert new_index.equals(index.droplevel(0))
+
+
+@pytest.mark.parametrize('level', [0, 1])
+@pytest.mark.parametrize('null_val', [np.nan, pd.NaT, None])
```

Comment:
we have a null fixture that does this (for the nans u r listing)

Reply:
Using the fixture `nulls_fixture` increases the number of nulls (and tests) by four, from 377 to 381.